### PR TITLE
Nodetool Arguments

### PR DESCRIPTION
**Pull request by chuckydev**
- Added the ability to pass custom NODETOOL_ARGS into all places that involve nodetool commands
- This includes the cassandra-snapshot and lifecycle hook commands
- The env var can be passed in from the previous feature of passing environment variables to the cassandra node itself as all nodetool commands are currently run from the node iteself.

### CODE REVIEW

File: cassandra-snapshot/pkg/nodetool/nodetool.go [hunk 1]
```diff
@@ -41,21 +45,19 @@ func New(kubeClientset *kubernetes.Clientset, restConfig *rest.Config) *Nodetool
...
 func (n *Nodetool) CreateSnapshot(snapshotTimestamp time.Time, keyspaces []string, pod *v1.Pod, snapshotCreationTimeout time.Duration) error {
 	snapshotName := strconv.FormatInt(snapshotTimestamp.Unix(), 10)
-	args := []string{"nodetool", "snapshot", "-t", snapshotName}
-	log.Infof("Creating Snapshot %s for pod %s and keyspaces %v", snapshotName, pod.Name, keyspaces)
 
-	for _, ks := range keyspaces {
-		args = append(args, ks)
-	}
+	log.Infof("Creating Snapshot %s for pod %s and keyspaces %v", snapshotName, pod.Name, keyspaces)
+	command := fmt.Sprintf("nodetool ${%s} snapshot -t %s -- %s", nodetoolArguments, snapshotName, strings.Join(keyspaces, " "))
```
Comment (Bez625) [inline cassandra-snapshot/pkg/nodetool/nodetool.go:50]: I'm probably missing something obvious, but where was the `--` in the original code?

**Review by Bez625 — CHANGES_REQUESTED** [review]

File: cassandra-snapshot/pkg/nodetool/nodetool.go (hunk 1 above)
Comment (chuckydev) [reply cassandra-snapshot/pkg/nodetool/nodetool.go:50]: https://docs.datastax.com/en/dse/5.1/dse-admin/datastax_enterprise/tools/nodetool/toolsSnapShot.html

It aligns with the cleanup commands aswell as following suggested structure from nodetool docs. The -- seperates the list of keyspaces from being mistaken as an arguement. Im not sure why it was emmitted originally, yet I had better results with the --

File: cassandra-snapshot/test/e2e/cassandra_builder.go [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+	"k8s.io/api/core/v1"
+	metaV1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"time"
+)
+
+var (
+	terminateImmediately = int64(0)
+)
+
+type CassandraBuilder struct {
```
[the remark this answers is not in the document]
Comment (chuckydev) [reply cassandra-snapshot/test/e2e/cassandra_builder.go:15]: This doesnt build a cluster. It builds a cassandra pod (not joined with any other pods, just simple cassandra instances). Happy with CassandraPodBuilder? I prefer CassandraBuilder personally.

File: cassandra-snapshot/test/e2e/cassandra_builder.go [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+	envVars []v1.EnvVar
+}
+
+func ACassandra() *CassandraBuilder {
+	return &CassandraBuilder{}
+}
+
+func (c *CassandraBuilder) WithDefaults() *CassandraBuilder {
```
[the remark this answers is not in the document]
Comment (chuckydev) [reply cassandra-snapshot/test/e2e/cassandra_builder.go:23]: Happy to do this. However it contradicts the removal of WithEnvironmentVariables. Im happy with "ACassandraWithDefaults" and a "WithoutEnvironmentVariables" defined

File: cassandra-snapshot/test/e2e/cassandra_builder.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (Bez625) [reply cassandra-snapshot/test/e2e/cassandra_builder.go:15]: I'd be happy with CassandraPodBuilder or PodBuilder. Strongly against CassandraBuilder as:
* It doesnt build a Cassandra.
* It's behaviour is inconsistent with the existing CassandraBuilder

File: cassandra-snapshot/test/e2e/cassandra_builder.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (Bez625) [reply cassandra-snapshot/test/e2e/cassandra_builder.go:23]: I didnt intend to suggest removing `WithEnvironmentVariables` - if you are referring to https://github.com/sky-uk/cassandra-operator/pull/254#discussion_r572897373 I meant removing the need to call `WithEnvironmentVariables` with an empty envVar list. 

E,g, have these methods:
* `APodWithDefaults` - return cluster builder with default env vars
* `APod`- return cluster builder with empty env vars. You could call this with `APod().WithEnvVars(x)` to set them

File: cassandra-snapshot/test/e2e/cassandra_builder.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (chuckydev) [reply cassandra-snapshot/test/e2e/cassandra_builder.go:15]: It does build a valid cassandra, just not an entire cluster or ring. I am happy with CassandraPodBuilder.

**Review by Bez625 — DISMISSED** [review]

**Pull request merged**